### DISCUSSION TITLE
No "Local O&M" in Fusion App - also cannot connect to Inverter

### DISCUSSION THREAD

**Discussion by bigretromike** [wlcrs/huawei_solar#816] [Setting up the integration] [answered]
Instalation is: 
SUN2000-8KTL-M1: V100R001C00SPC164
SDongleA-05: fw V100R001C00SPC119

In Commission menu I enabled unrestricted access.

By connecting to Home WiFi and nmap scan inverter I see tcp/502 being open but HA can't connect to it.
By connecting directly to build-in wifi and nmap scan Inverted I see tcp/6607 port being open but I don't have ability to connect HA directly, but modbus scanners don't pick up connection anyway.

I tried Enable(restricted) and add HA or/and PC IP to access but it no used. 

I don't know if this is because there is not "Local O&M" option to enable or I need to do something else.

HA is on different network than Inverter but there are connection between them, I can ping without issues inverter from HA.

I looked thru Compatibility Table but I dont see my FW.

**Comment by Roving-Ronin** [comment]
Your firmware and M1 setup shouold work fine. Just use the enable with unrestricted (IP) connecting to modbus, and your ID# is correct.
i.e. Should be 1 (unless the installer fiddled with it).

**Reply by bigretromike** [reply]
I would thing the same but it's no go, @Roving-Ronin 

I tried CAS Modbus Scan application to find proper ID but all of them 0-255 timeout. I'm in same wifi network. I can see from nmap that 502 is open. But no matter what way I try to access data I cannot get anything.

I enabled (unrestricted) mode in configuration, but it looks like It wont allow me to access anything. I tried with restricted mode and added my ip, same results. 

I tried pure pyModbusTCP
```

from pyModbusTCP.client import ModbusClient
c = ModbusClient(host="10.9.9.111", port=502, unit_id=1, auto_open=True, auto_close=False, timeout=10)
c.open()
if c.is_open:
    regs = c.read_holding_registers(0, 15)
    if regs:
        print(regs)
    else:
        print("read error")
    c.close()
```
gets "read error" because of null read. I tried all auto_open & auto_close combinations.

tried dedicated lib for this unit:
```

from sun2000_modbus import inverter
from sun2000_modbus import registers

inverter = inverter.Sun2000(host='10.9.9.111', port=502, unit=1, wait=10)
inverter.connect()
if inverter.isConnected():
    input_power = inverter.read_formatted(registers.InverterEquipmentRegister.Model)
    print(input_power)
```
and got: pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] ModbusTcpClient([10.9.9.111:502](http://10.9.9.111:502/)): Connection unexpectedly closed 0.000000 seconds into read of 8 bytes without response from unit before it closed connection

For me it looks like Modbus is working (opened 502/tcp port) but wont allow me to access any data.

Once again I can confirm in SolarFusion that Modbus TCP Connection is "Enable (unrestricted)"  and dongle parameter settings > start address from automatic address allocation: 1. Also Communication configuration > parallel system communication parameter settings > parallel communication mode > NA

**Comment by bigretromike** [comment, marked as answer]
I contacted Huawei Support.

They told me to upgrade Sdongle I tried to upgrade to V100R001C00SPC135 but it failed saying its not for this device.
But I tried one lower which was V100R001C00SPC133, after that I was able to connect with Python and CASModbusScanner.

Now I can add it to HA.

Maybe someone will benefit from this information, thanks.

**Reply by Roving-Ronin** [reply]
Wow yes you had VERY old firmware on sDongle. No flame, but this is one reason why there's the Wiki, with a section on firmware revisions:

https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions#three-phase---sun2000-m1-hybrid

I would suggest that you skip past the legacy firmware and move to v2, where the new features and functions are being implemented.
As per Wiki other users are on:

8KTL-M1: V100R001C00SPC165 (SPC164 is close enough)
sDongle: V200R022C10SPC118

Also check your MBUS (aka Modbus) firmware, it doesn't get updated as part of the inverter firmware, you have to specifically update it seperately. SPC335 is the current revision, however depending upon where you live the local Huawei support may block you to SPC330 (i.e. Australia does this based on their views).

**Reply by bigretromike** [reply]
> I would suggest that you skip past the legacy firmware and move to v2

How would I skip legacy firmware ?

**Reply by Roving-Ronin** [reply]
In SolarFusion Portal, when updating the firmware (and MBUS) simply choose the newer V200 firmware. In this case V200R022C10SPC118 (or newer). I have V200R022C10SPC118 running on my M1 though and no issues.

**Reply by bigretromike** [reply]
oh, I always though V200 is version of device and not firmware 👍

**Reply by bigretromike** [reply]
Just installed V200R022C10SPC118 without issue
Inverter on V100R001C00SPC165
Optimizers on V100R002C10SPC019
MBus V100R001C00SPC335

Working without issues
